### ISSUE TITLE
Testrpc fails to initialize when .provider or .server receive an account array

### ISSUE THREAD

**Issue by sogoiii** [ConsenSys-archive/ganache-cli-archive#277]
When you provide an object to `provider` or `server` that contains an `accounts` array, you get `AssertionError: Invalid index` error shown below. Providing the key `secretKey` does not resolve the issue.

```
/Users/myself/Documents/ethereum/github/testrpc/node_modules/solc/soljson.js:1
(function (exports, require, module, __filename, __dirname) { var Module;if(!Module)Module=(typeof Module!=="undefined"?Module:null)||{};var moduleOverrides={};for(var key in Module){if(Module.hasOwnProperty(key)){moduleOverrides[key]=Module[key]}}var ENVIRONMENT_IS_WEB=typeof window==="object";var ENVIRONMENT_IS_WORKER=typeof importScripts==="function";var ENVIRONMENT_IS_NODE=typeof process==="object"&&typeof require==="function"&&!ENVIRONMENT_IS_WEB&&!ENVIRONMENT_IS_WORKER;var ENVIRONMENT_IS_SHELL=!ENVIRONMENT_IS_WEB&&!ENVIRONMENT_IS_NODE&&!ENVIRONMENT_IS_WORKER;if(ENVIRONMENT_IS_NODE){if(!Module["print"])Module["print"]=function print(x){process["stdout"].write(x+"\n")};if(!Module["printErr"])Module["printErr"]=function printErr(x){process["stderr"].write(x+"\n")};var nodeFS=require("fs");var nodePath=require("path");Module["read"]=function read(filename,binary){filename=nodePath["normalize"](filename);var ret=nodeFS["rea
AssertionError: Invalid index
    at /Users/myself/Documents/ethereum/github/testrpc/node_modules/hdkey/lib/hdkey.js:86:5
    at Array.forEach (native)
    at HDKey.derive (/Users/myself/Documents/ethereum/github/testrpc/node_modules/hdkey/lib/hdkey.js:78:11)
    at EthereumHDKey.derivePath (/Users/myself/Documents/ethereum/github/testrpc/node_modules/ethereumjs-wallet/hdkey.js:37:32)
    at StateManager.createAccount (/Users/myself/Documents/ethereum/github/testrpc/lib/statemanager.js:150:28)
    at Array.map (native)
    at StateManager.initialize (/Users/myself/Documents/ethereum/github/testrpc/lib/statemanager.js:71:33)
    at new GethApiDouble (/Users/myself/Documents/ethereum/github/testrpc/lib/subproviders/geth_api_double.js:23:14)
    at Object.module.exports [as provider] (/Users/myself/Documents/ethereum/github/testrpc/lib/provider.js:29:23)
    at Object.<anonymous> (/Users/myself/Documents/ethereum/b9Course/evnts/makingNewAccounts.js:48:26)
```

```javascript
const Web3 = require('web3')
const TestRPC = require('ethereumjs-testrpc')
const web3 = new Web3()
web3.setProvider(TestRPC.provider({
  port: 9999,
  accounts: [
    { balance: '0x12' },
    { balance: '0x13' }
  ]
}))

web3.eth.getAccounts((err, result) => {
  if (err) { console.log(err) }
  console.log(result)
})
```

 

## Expected Behavior

Firstly, I expect the code to not error out. But from the code snippet above, I expect an array of 2 users to be console logged. 

## Current Behavior

currently getting the error stack show:

```
/Users/myself/Documents/ethereum/github/testrpc/node_modules/solc/soljson.js:1
(function (exports, require, module, __filename, __dirname) { var Module;if(!Module)Module=(typeof Module!=="undefined"?Module:null)||{};var moduleOverrides={};for(var key in Module){if(Module.hasOwnProperty(key)){moduleOverrides[key]=Module[key]}}var ENVIRONMENT_IS_WEB=typeof window==="object";var ENVIRONMENT_IS_WORKER=typeof importScripts==="function";var ENVIRONMENT_IS_NODE=typeof process==="object"&&typeof require==="function"&&!ENVIRONMENT_IS_WEB&&!ENVIRONMENT_IS_WORKER;var ENVIRONMENT_IS_SHELL=!ENVIRONMENT_IS_WEB&&!ENVIRONMENT_IS_NODE&&!ENVIRONMENT_IS_WORKER;if(ENVIRONMENT_IS_NODE){if(!Module["print"])Module["print"]=function print(x){process["stdout"].write(x+"\n")};if(!Module["printErr"])Module["printErr"]=function printErr(x){process["stderr"].write(x+"\n")};var nodeFS=require("fs");var nodePath=require("path");Module["read"]=function read(filename,binary){filename=nodePath["normalize"](filename);var ret=nodeFS["rea
AssertionError: Invalid index
    at /Users/myself/Documents/ethereum/github/testrpc/node_modules/hdkey/lib/hdkey.js:86:5
    at Array.forEach (native)
    at HDKey.derive (/Users/myself/Documents/ethereum/github/testrpc/node_modules/hdkey/lib/hdkey.js:78:11)
    at EthereumHDKey.derivePath (/Users/myself/Documents/ethereum/github/testrpc/node_modules/ethereumjs-wallet/hdkey.js:37:32)
    at StateManager.createAccount (/Users/myself/Documents/ethereum/github/testrpc/lib/statemanager.js:150:28)
    at Array.map (native)
    at StateManager.initialize (/Users/myself/Documents/ethereum/github/testrpc/lib/statemanager.js:71:33)
    at new GethApiDouble (/Users/myself/Documents/ethereum/github/testrpc/lib/subproviders/geth_api_double.js:23:14)
    at Object.module.exports [as provider] (/Users/myself/Documents/ethereum/github/testrpc/lib/provider.js:29:23)
    at Object.<anonymous> (/Users/myself/Documents/ethereum/b9Course/evnts/makingNewAccounts.js:48:26)
```

## Steps to Reproduce (for bugs)

1. Create file called `showAccounts.js` with given code snippet.
2. install `web3` and `ethereumjs-testrpc`
3. run `node showAccounts.js`
4. you will get error.

## Context

Trying to define the balance an account has programatically. 

## Your Environment

* Version used: node 7.7.2, npm 4.1.2, testrpc 3.0.3.
* Operating System and version: OSX El Capitan

**Comment by travs** [comment]
Identical error on Windows 7, Node v6.9.1, testrpc 3.0.3

Your PR #278 worked for me as well.

**Comment by benjamincburns** [comment]
I think this was fixed by #278, but we just forgot to close it. Please reopen if this is a mistake.
